environment.go

```diff
@@ -164,6 +150,37 @@ nextExt:
 		default:
 		}
 	}
+	// This is here as we could've not run go mod tidy due to a replace being the only extension
+	err = env.execGoModTidy(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// write the main module file to temporary folder
+	// we do this last so we get the needed versions from all the replacements and extensions instead of k6 if possible
+	mainPath := filepath.Join(tempFolder, "main.go")
+	log.Printf("[INFO] Writing main module: %s", mainPath)
+	err = ioutil.WriteFile(mainPath, buf.Bytes(), 0o600)
+	if err != nil {
+		return nil, err
+	}
+
+	// building with the default main repo
+	if b.K6Repo == "" && env.k6Version != "" {
+		// don't actually specify k6 version if it wasn't specified
+		// the extension(s) will require a version that they work with either way
+		// but if it was let specify it now
+		// all the previous steps should've worked so far as the extensions themselves would've required
+		// specific versions to begin with and they should work with those versions
+		err = env.execGoModRequire(ctx, k6ModulePath, env.k6Version)
```

Comment:
> I would tidy up the comment a bit, if you don't mind :) 

:sob:. What specifically do you not like?

> But how would these changes work if two extensions specify different k6 versions?

The highest one between them - this does not in any way change how go modules work. Just before this changes we in fact added `k6@latest` as one of the version ... which likely was going to be the *highest* one requested by any dependancy and so it will be used. 

All the remaining changes are basically because I still need to write the `main.go` and as the commit messages try to say that means that the next `go mod tidy` will get a verison of k6. If nothing so far has requested one ... it will get the latest which is again what we are trying to not do. 

So all the restructuring is so that we don't just request as dependancy the latest version of k6  and still have all the features working :crossed_fingers: 

> ... which fails because xk6-redis isn't compatible with latest k6 (undefined: common.GetRuntime), as expected.

There is no way to fix that except `k6` to start bumping the major version on each breaking change - which isn't happening anytime soon as you know. 

I would argue that also building *multiple* extensions in the same binary is a lot less likely than people needing *one* extension. And this in no way shape or form makes building multiple extensions harder - again it just doesn't enforce bumping the k6 version to latest each time you build and haven't provided k6 version on the cli. 

>env.k6Version is empty (i.e. not this execution path, just asking here because of the comment)

env.k6Version is empty if you do `xk6 build --with ....` but is `something` in `xk6 build something --with=...`. Am I missing something. The comment above more or less is saying what I have been saying above - extensions will provide their required version of k6 - we shouldn't add one if it wasn't specified on the cli, if it was - we obviously should ;). And the the normal go module algorithm will pick the highest request version as with any other dependancy in go.

Reply:
OK, understood.

> env.k6Version is empty if you do `xk6 build --with ....` but is `something` in `xk6 build something --with=...`.

Yeah, I just wanted to know what would happen in the other scenario (when it's empty), but made the comment here because of your comment.

> What specifically do you not like?

Some punctuation, grammar fixes and less repetition would help :smile:. Maybe?

```suggestion
		// Only require a specific k6 version if provided. Otherwise extensions
		// will require a version they depend on, and Go's module resolution
		// algorithm will choose the highest one among all extensions.
		err = env.execGoModRequire(ctx, k6ModulePath, env.k6Version)
```